ambry-account/src/main/java/com/github/ambry/account/AccountUtils.java

```diff
@@ -100,69 +101,74 @@ public static int compareAccounts(Collection<Account> accountsInPrimary, Collect
 
       StringBuilder accountsInfo = new StringBuilder();
 
-      if (!accountsMissingInSecondary.isEmpty()) {
-        accountsInfo.append("[");
-        for (Account account : accountsMissingInSecondary) {
-          accountsInfo.append(AccountCollectionSerde.accountToJsonNoContainers(account).toString()).append(",");
+      try {
+        if (!accountsMissingInSecondary.isEmpty()) {
+          accountsInfo.append("[");
+          for (Account account : accountsMissingInSecondary) {
+            accountsInfo.append(new String(AccountCollectionSerde.serializeAccountsInJsonNoContainers(account)))
+                .append(",");
+          }
+          accountsInfo.append("]");
+          logger.warn("Accounts found in primary and absent in secondary = {}", accountsInfo.toString());
+          mismatchCount += accountsMissingInSecondary.size();
         }
-        accountsInfo.append("]");
-        logger.warn("Accounts found in primary and absent in secondary = {}", accountsInfo.toString());
-        mismatchCount += accountsMissingInSecondary.size();
-      }
 
-      if (!accountsDifferentInSecondary.isEmpty()) {
-        accountsInfo.setLength(0);
-        accountsInfo.append("[");
-        for (Account account : accountsDifferentInSecondary) {
-          accountsInfo.append("{Account = ")
-              .append(account.toString())
-              .append(", primary = ")
-              .append(AccountCollectionSerde.accountToJsonNoContainers(account).toString())
-              .append(", secondary = ")
-              .append(AccountCollectionSerde.accountToJsonNoContainers(secondaryAccountMap.get(account.getName()))
-                  .toString());
+        if (!accountsDifferentInSecondary.isEmpty()) {
```

Comment:
Minor: It's strange that github show space before `if`.

Reply:
this is not space, this is the new indentation since I add a try-catch block to surround the if statements.